### PR TITLE
Angr remove bare functions Part-4

### PR DESCRIPTION
**Pull request by Adersh97**

### CODE REVIEW

**Comment by rhelmot** [comment]
Doesn't unittest not support the generator style tests? Shouldn't we remove the generators before we convert them to unittest?

File: tests/test_echo.py [hunk 1]
```diff
@@ -22,8 +32,10 @@ def _run_echo_haha(self,arch):
         # Need to dump by path because the program closes stdout
         assert pg.deadended[0].posix.stdout.concretize() == [b'haha\n']
 
-    def test_run_x86_64(self):
-        self._run_echo_haha('x86_64')
+    def test_echo_haha(self):
+        for arch in target_arches:
+            yield self.run_echo_haha, arch
```
Comment (twizmwazin) [inline tests/test_echo.py:37]: This generator needs to be refactored into a normal function

**Review by twizmwazin — COMMENTED** [review]
Other than that one generator, LGTM!

File: tests/test_echo.py [hunk 2]
```diff
@@ -22,8 +32,10 @@ def _run_echo_haha(self,arch):
         # Need to dump by path because the program closes stdout
         assert pg.deadended[0].posix.stdout.concretize() == [b'haha\n']
 
-    def test_run_x86_64(self):
-        self._run_echo_haha('x86_64')
+    def test_echo_haha(self):
+        for arch in target_arches:
```
Comment (twizmwazin) [inline tests/test_echo.py:36]: It looks like you reverted some of the changes here when fixing a conflict

Comment (Adersh97) [reply tests/test_echo.py:36]: It is still the same, I thought it looked clean to run it through iteration of versions, I will revert it back to how it was

**Review by twizmwazin — APPROVED** [review]

**Pull request merged**